rasa/core/migrator.py

```diff
@@ -0,0 +1,214 @@
+        maximum_timestamp: Optional[float] = None,
+    ):
+        """
+        Args:
+            endpoints_path: Path to the endpoints file used to configure the event
+                broker and tracker store. If `None`, the default path ('endpoints.yml')
+                is used.
+            requested_conversation_ids: List of conversation IDs requested to be
+                processed.
+            minimum_timestamp: Minimum timestamp of events that are published.
+                If `None`, apply no such constraint.
+            maximum_timestamp: Maximum timestamp of events that are published.
+                If `None`, apply no such constraint.
+
+        """
+        self.endpoints_path = endpoints_path
+        self.tracker_store = tracker_store
+        self.event_broker = event_broker
+        self.requested_conversation_ids = requested_conversation_ids
+        self.minimum_timestamp = minimum_timestamp
+        self.maximum_timestamp = maximum_timestamp
+
+    def publish_events(self) -> int:
+        """Publish events in a tracker store using an event broker.
+
+        Exits if the publishing of events is interrupted due to an error. In that case,
+        the CLI command to continue the export where it was interrupted is printed.
+
+        """
+        events = self._fetch_events_within_time_range()
+
+        cli_utils.print_info(
+            f"Selected {len(events)} events for publishing. Ready to go 🚀"
+        )
+
+        published_events = 0
+        current_timestamp = None
+
+        for event in tqdm(events, "events"):
+            # noinspection PyBroadException
+            try:
+                body = {"sender_id": event["sender_id"]}
+                body.update(event)
+                self.event_broker.publish(body)
+                published_events += 1
+                current_timestamp = event["timestamp"]
+            except Exception as e:
+                logger.exception(e)
+                raise PublishingError(current_timestamp)
+
+        self.event_broker.close()
+
+        return published_events
+
+    def get_conversation_ids_to_process(self) -> Set[Text]:
+        """Get conversation IDs that are good for processing.
+
+        Finds the intersection of events that are contained in the tracker store with
+        those events requested as a command-line argument.
+
+        Prints an error and if no conversation IDs are found in the tracker, or if no
+        overlap is found between those contained in the tracker and those requested
+        by the user.
+
+        Returns:
+            Conversation IDs that are both requested and contained in the tracker
+            store. If no conversation IDs are requested, all conversation IDs in the
+            tracker store are returned.
+
+        """
+        conversation_ids_in_tracker_store = set(self.tracker_store.keys())
+
+        if not conversation_ids_in_tracker_store:
+            raise NoConversationsInTrackerStore(
```

Comment:
let's use the `Error` suffix https://www.python.org/dev/peps/pep-0008/#exception-names